grouper/user_password.py

```diff
@@ -0,0 +1,48 @@
+from grouper.models.counter import Counter
+from grouper.models.user_password import UserPassword
+from sqlalchemy.exc import IntegrityError
+
+
+class PasswordAlreadyExists(Exception):
+    pass
+
+
+class PasswordDoesNotExist(Exception):
+    pass
+
+
+def add_new_user_password(session, password_name, password, user_id):
+    # type: Session, str, str, int -> None
+    """Add the new user password specified.
+
+    Args:
+        session(grouper.models.base.session.Session): database session
+        password_name(str): name of the password to be added
+        password(str): the (plaintext) password to be added
+        user_id(int): the id of the user to add this password to
+    """
+    p = UserPassword(name=password_name, user_id=user_id)
+    p.password = password
+    p.add(session)
+    try:
+        session.commit()
+    except IntegrityError:
+        raise PasswordAlreadyExists()
+    Counter.incr(session, "updates")
+
+
+def delete_user_password(session, password_name, user_id):
+    # type: Session, str, int -> None
+    """Delete the specified UserPassword.
+
+    Args:
+        session(grouper.models.base.session.Session): database session
+        password_name: the name of the password to delete
+        user_id: the user whose password is being deleted
+    """
+    p = session.query(UserPassword).filter_by(name=password_name, user_id=user_id).scalar()
+    if not p:
+        raise PasswordDoesNotExist()
+    p.delete(session)
```

Comment:
In general, I try to avoid deleting objects from the database for audit/recoverability reasons. But @herb feel free to disagree. 

Perhaps instead mark the password as inactive and rename it?